src/ui/utils/timeline.ts

```diff
@@ -22,6 +23,14 @@ export function getPixelDistance({
   return Math.abs((toPos - fromPos) * overlayWidth);
 }
 
+export function startTimeForFocusRegion(focusRegion: FocusRegion) {
```

Comment:
These are just wrappers that know about checking prefs, which keeps the logic contained.

Reply:
I like that we're encapsulating this logic in a single place.

I'm concerned though that there is no static analysis checks to prevent people from reading `focusRegion.start` or `focusRegion.startTime` directly.

I wonder if it would be worth making this type opaque so that people _couldn't_ directly interact with those attributes (without a type cast or ts-ignore)? That might help avoid references from leaking in.

Reply:
Good idea. I've made the `FocusRegion` type empty, and made an `UnsafeFocusRegion` which you can cast to if you need to directly access fields on it.

Reply:
Thanks! 🙇🏼